inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/config/model/instrumentation/scope/MethodMatcherSettings.java

```diff
@@ -0,0 +1,50 @@
+package rocks.inspectit.oce.core.config.model.instrumentation.scope;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+import javax.validation.constraints.NotNull;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Data container for settings which will be used as basis for the {@link rocks.inspectit.oce.core.instrumentation.config.model.InstrumentationScope}
+ * and its method matcher.
+ * <p>
+ * An instance of this class represents a matcher which is matching a set of methods.
+ */
+@Data
+@NoArgsConstructor
+@EqualsAndHashCode(callSuper = true)
+public class MethodMatcherSettings extends NameMatcherSettings {
+
+    /**
+     * Enum for access modifiers which can be used in the instrumentation configuration.
+     */
+    public enum AccessModifier {
+        PUBLIC, PROTECTED, PACKAGE, PRIVATE
+    }
+
+    /**
+     * Whether the method is a constructor or not.
+     */
+    @NotNull
+    private Boolean isConstructor = false;
+
+    /**
+     * Whether the method is synchronized or not. If this property is `null` the synchronize keyword will be ignored.
+     */
+    private Boolean isSynchronized;

```

Comment:
At first, I was thinking this is really not necessary, but then I understood that it would be quite easy to instrument all synchronized methods with some wild cards, so quite handy..